### PR TITLE
Extra gosec checks

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
This pr added extra redundancy to what we already have with golangci-lint-action such as:
 - Explicit linter check. This check is focused on gosec only and not mixed with any other thing.
 - On each update and by schedule upload sarif report  to the github https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning

### Screenshot/screencast of this PR
n/a

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/SVPI-265

### How to test this PR?
n/a

### CODE REVIEW

File: .github/workflows/linter.yaml [hunk 1]
```diff
@@ -31,3 +31,14 @@ jobs:
...
           # skip-build-cache: true
+  gosec:
+    runs-on: ubuntu-latest
+    env:
+      GO111MODULE: on
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v3
+      - name: Run Gosec Security Scanner
+        uses: securego/gosec@master
```
Comment (sparkoo) [inline .github/workflows/linter.yaml:42]: is it good idea to run from master?

File: .github/workflows/sec.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: "Security Scan"
+
+# Run workflow each time code is pushed to your repository and on a schedule.
+# The scheduled workflow runs every at 00:00 on Sunday UTC time.
+on:
+  push:
```
Comment (sparkoo) [inline .github/workflows/sec.yaml:6]: does this run on push or this action is run weekly?

File: .github/workflows/sec.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  schedule:
+    - cron: '0 0 * * 0'
```
Comment (sparkoo) [inline .github/workflows/sec.yaml:8]:
```suggestion
    - cron: '0 0 * * 0' # every Sunday 00:00 UTC
```

File: .github/workflows/sec.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+name: "Security Scan"
+
+# Run workflow each time code is pushed to your repository and on a schedule.
+# The scheduled workflow runs every at 00:00 on Sunday UTC time.
```
Comment (sparkoo) [inline .github/workflows/sec.yaml:4]: I would just remove this whole comment and write inline comment to cron time

File: .github/workflows/linter.yaml [hunk 5]
```diff
@@ -31,3 +31,14 @@ jobs:
 
           # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
           # skip-build-cache: true
+  gosec:
```
Comment (sparkoo) [inline .github/workflows/linter.yaml:34]: what level of security issue makes the whole check red?

Comment (metlos) [reply .github/workflows/linter.yaml:34]: I think these just might be caused the fact that you're using the latest greatest version of the github action, while our lint task pins the version of golang-lint. What about trying to change this https://github.com/redhat-appstudio/service-provider-integration-operator/blob/315765437588437b7cc9fb2a9ce1efd6c7764ecd/.github/workflows/linter.yaml#L18 to `latest` and see if we get the same new warnings?

**Review by metlos — COMMENTED** [review]
I would personally rather modify our existing linting infra than to add another github action that basically duplicates what we already should have done in the linter.
If there is no other way, then I'm ok with this but I'd like to first try to not have to have duplicate work done.

**Review by sparkoo — APPROVED** [review]

**Pull request merged**